src/libstore/build/derivation-goal.cc

```diff
@@ -1019,7 +1020,45 @@ void DerivationGoal::buildDone()
 }
 
 void DerivationGoal::resolvedFinished() {
-    done(BuildResult::Built);
+    assert(resolvedDrv);
+
+    // If the derivation was originally a full `Derivation` (and not just
+    // a `BasicDerivation`, we must retrieve it because the `staticOutputHashes`
+    // will be wrong otherwise
+    Derivation fullDrv = *drv;
+    if (auto upcasted = dynamic_cast<Derivation *>(drv.get()))
+        fullDrv = *upcasted;
+
+    auto originalHashes = staticOutputHashes(worker.store, fullDrv);
+    auto resolvedHashes = staticOutputHashes(worker.store, *resolvedDrv);
```

Comment:
As an aside I find the "hash" terminology confusing since it's not really clear what it is a hash of... We should find a better name.

Reply:
I agree that the terminology isn't really ideal. I couldn't think of anything better though…

Maybe we could just manipulate the `DrvOutput` `(hash, outputName)`, so that we don't have to name this precise bit which doesn't make much sense by itself